include/hipSYCL/sycl/libkernel/host/builtins.hpp

```diff
@@ -528,6 +529,56 @@ HIPSYCL_BUILTIN T __hipsycl_clamp(T x, T minval, T maxval) noexcept {
   return std::min(std::max(x, minval), maxval);
 }
 
+template<class T, std::enable_if_t<std::is_integral_v<T>,int> = 0>
+inline T __fallback_clz(T x) noexcept {
```

Comment:
If starting with `__`, we should probably prefix with `__hipsycl`. Underscore names are reserved in C++ for the compiler implementation, and we'd like to avoid trouble with the compiler :-) But we could also just not have `__` here, as it's just a host helper function, not an actual builtin.